cli/main.rs

```diff
@@ -1147,7 +1111,12 @@ pub fn main() {
   if !flags.v8_flags.is_empty() {
     init_v8_flags(&*flags.v8_flags);
   }
-  init_logger(flags.log_level);
 
-  unwrap_or_exit(tokio_util::run_basic(get_subcommand(flags)));
+  // when this flag is set to `true`, the process will log specific diagnostic
+  // debug information,
+  let lsp_debug_flag = Arc::new(AtomicBool::new(false));
```

Comment:
It's unfortunate that this `lsp_debug_flag` has leaked all the way to top-level of the program and many signatures have to be modified to pass it around.

What if you use lazy_static to have a global `Arc<AtomicBool>` in `cli/logger.rs`? Then `cli/lsp/language_server.rs` could reference that directly without having to pass it around like this. I think practically this `lsp_debug_flag` is global anyway.

Reply:
Yeah, that makes more sense. I will try that.